piccolo/apps/schema/commands/generate.py

```diff
@@ -251,28 +267,134 @@ async def get_table_schema(
+    )
+    table_schema = await get_table_schema(
+        table_class=Schema, tablename=tablename, schema_name=schema_name
+    )
+
+    columns: t.Dict[str, Column] = {}
+
+    for pg_row_meta in table_schema:
+        data_type = pg_row_meta.data_type
+        column_type = COLUMN_TYPE_MAP.get(data_type, None)
+        column_name = pg_row_meta.column_name
+        if not column_type:
+            output_schema.warnings.append(
+                f"{tablename}.{column_name} ['{data_type}']"
+            )
+            column_type = Column
+
+        kwargs: t.Dict[str, t.Any] = {
+            "null": pg_row_meta.is_nullable == "YES",
+            "unique": constraints.is_unique(column_name=column_name),
+        }
+
+        if constraints.is_primary_key(column_name=column_name):
+            kwargs["primary_key"] = True
+            if column_type == Integer:
+                column_type = Serial
+
+        if constraints.is_foreign_key(column_name=column_name):
+            fk_constraint_table = constraints.get_foreign_key_constraint_name(
+                column_name=column_name
+            )
+            column_type = ForeignKey
+            constraint_table = await get_foreign_key_reference(
+                table_class=Schema,
+                constraint_name=fk_constraint_table.name,
+                constraint_schema=fk_constraint_table.schema,
+            )
+            if constraint_table.name:
+                kwargs["references"] = await create_table(
+                    Schema,
+                    constraint_table.name,
+                    constraint_table.schema,
+                    output_schema,
+                )
+            else:
+                kwargs["references"] = ForeignKeyPlaceholder
+
+        output_schema.imports.append(
+            "from piccolo.columns.column_types import " + column_type.__name__
+        )
+
+        if column_type is Varchar:
+            kwargs["length"] = pg_row_meta.character_maximum_length
+
+        column = column_type(**kwargs)
+
+        serialised_params = serialise_params(column._meta.params)
+        for extra_import in serialised_params.extra_imports:
+            output_schema.imports.append(extra_import.__repr__())
+
+        columns[column_name] = column
+
+    table = create_table_class(
+        class_name=_snake_to_camel(tablename),
+        class_kwargs={"tablename": get_table_name(tablename, schema_name)},
+        class_members=columns,
+    )
+    output_schema.tables.append(table)
+    return table
+
+
+async def get_output_schema(
+    schema_name: str = "public",
+    tablenames: t.Optional[t.List[str]] = None,
+    exclude: t.Optional[t.List[str]] = None,
+) -> OutputSchema:
+    """
+
+    :param schema_name:
+    name of the schema
+    :param tablenames:
+    optional list of table names. Only creates the specifed tables.
+    :param exclude:
+    optional list of table names. excludes the specified tables.
+    :return:
+    OutputSchema
```

Comment:
We should try and indent these to make them more readable and consistent with the rest of the code base.

Something like:

```python
    """
    :param schema_name:
        Name of the schema
    :param tablenames:
        Optional list of table names. Only creates the specifed tables.
    :param exclude:
        Optional list of table names. Excludes the specified tables.
    :returns:
        OutputSchema
    """
```